### ISSUE TITLE
Improve external language documentation

### ISSUE THREAD

**Issue by Athas** [HIPERFIT/futhark#37]
While the internal language and compiler design changes frequently, the external language is fairly stable.  We should improve the documentation (`doc/futhark.tex`) in places where it is currently insufficient.

Of course, another problem is that writing documentation as a LaTeX document intended to be compiled into a PDF seems horribly old-fashioned.  What do the cool kids do nowadays?  We still want to be able to have decent math support, probably.

**Comment by RasmusWL** [comment]
The cool kids use [readthedocs](https://readthedocs.org/). For LaTeX math we could use [MathJax](https://www.mathjax.org).

**Comment by Athas** [comment]
If we use readthedocs, then we still need to decide between Restructured Text and Markdown.

**Comment by Athas** [comment]
It basically seems that there is no reason to use Markdwn, so reStructuredText (as it is apparently supposed to be capitalised) it will be.

**Comment by kfl** [comment]
If it matters I would recommend markdown over reST.  But I think it is mostly a person preference. (I find the reST syntax rather ugly and non-intuitive).

**Comment by Athas** [comment]
Is that recommendation also for writing manual-sized documents (as in, maybe a hundred pages or more)?

**Comment by kfl** [comment]
I have not used markdown for one large document with hundreds of pages, but I have  switched to writing (almost) all my teaching notes in markdown. And I have also converted [some notes](http://www.diku.dk/~kflarsen/gd-2012/trees.html) from reST to markdown.

I can no longer remember exactly why I found reST to be so ugly, maybe it was also that pandoc had(?) slightly better support for markdown. To start with, it is fairly risk-less to experiment with both syntaxes as pandoc can convert one to the other. In general, I just recommend to use pandoc I think :wink:
